### ISSUE TITLE
Не стартует система на NAS WD MyBookLive

### ISSUE THREAD

**Issue by startsevsa** [pihel/cash#45]
Пробовал запустить на локальном сетевом хранилище - не работает (выдает ошибку, что не нашелся sqlite драйвер).
Мои собственные скрипты работают с этой СУБД замечательно - но через PDO.
Информацию о сервере (phpinfo) прилагаю.

P.S.: на денвере все ест-но работает нормально.

![screenshot__2015_09_14__10_59](https://cloud.githubusercontent.com/assets/14212283/9851972/29adf374-5b15-11e5-8e25-072691c5eb56.png)
![screenshot__2015_09_14__19_16](https://cloud.githubusercontent.com/assets/14212283/9851973/29ae5102-5b15-11e5-94b7-97421137de1e.png)

**Comment by pihel** [comment]
Добрый день,
Бухгалтерия не работает через драйвер pdo, только через нативный sqlite3 http://php.net/manual/ru/book.sqlite3.php

![sqlite3](https://cloud.githubusercontent.com/assets/860258/9852941/a8850338-5b09-11e5-9114-d91bcf7aa64f.png)

Но у меня нет данных, что чаще встречается: pdo_sqlite или нативные драйвера. Если pdo чаще, то есть смысл переделать.

**Comment by startsevsa** [comment]
А если, как вариант, не переписывать а копию сделать - т.е. в зависимости от драйвера вызывать соответствующие методы класса... ?
Или перегрузить методы для класса под PDO ?

**Comment by pihel** [comment]
Да, не нужно переписывать, нужно создать новый класс тут https://github.com/pihel/cash/tree/master/lib/db (Сейчас есть sqlite3, mysqli, odbc) и добавить настройки тут https://github.com/pihel/cash/blob/master/lib/init.php

Но сейчас, если это не шаред хостинг, быстрей подключить драйвер sqlite3.

**Comment by startsevsa** [comment]
Ну, хостинг не шаред, но это ручные ковыряния в NAS-е - т.е. при каждом обновлении микропрограммы нужно будет танцевать долго с бубном (и помнить всю последовательность танца).
А это конфигурация плагина, который подключается "из коробки"

**Comment by pihel** [comment]
Драйвер sqlite3 поставляется из коробки с php 5.3 (http://php.net/manual/ru/sqlite3.installation.php), драйвер pdo с 5.1 . С выхода 5.3 уже прошло более 6 лет.
Думаю, лучше было добавить поддержку pdo, как более распространенную, с самого начала, но на данном этапе быстро решить эту проблему не смогу. 
Запишу в todo и сделаю, когда будет время :)

**Comment by startsevsa** [comment]
да, на NAS у меня PHP Version 5.2.6-1+lenny9 :(
